e2e/nomostest/registryproviders/artifact_registry.go

```diff
@@ -0,0 +1,211 @@
+	var err error
+	authCmd := a.shell.Command("gcloud", "auth", "print-access-token")
+	loginCmd := a.shell.Command("crane", "auth", "login",
+		"-u", "oauth2accesstoken", "--password-stdin",
+		a.registryHost())
+	loginCmd.Stdin, err = authCmd.StdoutPipe()
+	if err != nil {
+		return fmt.Errorf("creating STDOUT pipe: %w", err)
+	}
+	if err := loginCmd.Start(); err != nil {
+		return fmt.Errorf("starting login command: %w", err)
+	}
+	if err := authCmd.Run(); err != nil {
+		return fmt.Errorf("running print-access-token command: %w", err)
+	}
+	if err := loginCmd.Wait(); err != nil {
+		return fmt.Errorf("waiting for login command: %w", err)
+	}
+	return nil
+}
+
+// Setup performs setup
+func (a *ArtifactRegistryOCIProvider) Setup() error {
+	if err := a.registryLogin(); err != nil {
+		return err
+	}
+	return a.createRepository()
+}
+
+// PushURL returns a URL for pushing images to the remote registry.
+// name refers to the repo name in the format of <NAMESPACE>/<NAME> of RootSync|RepoSync.
+func (a *ArtifactRegistryOCIProvider) PushURL(name string) (string, error) {
+	return fmt.Sprintf("%s/%s", a.repositoryAddress(), name), nil
+}
+
+// SyncURL returns a URL for Config Sync to sync from using OCI.
+// name refers to the repo name in the format of <NAMESPACE>/<NAME> of RootSync|RepoSync.
+func (a *ArtifactRegistryOCIProvider) SyncURL(name string) string {
+	return fmt.Sprintf("%s/%s", a.repositoryAddress(), name)
+}
+
+// ArtifactRegistryHelmProvider provides methods for interacting with the test registry-server
+// using the helm-sync interface.
+type ArtifactRegistryHelmProvider struct {
+	ArtifactRegistryProvider
+}
+
+func (a *ArtifactRegistryHelmProvider) registryLogin() error {
+	var err error
+	authCmd := a.shell.Command("gcloud", "auth", "print-access-token")
+	loginCmd := a.shell.Command("helm", "registry", "login",
+		"-u", "oauth2accesstoken", "--password-stdin",
+		a.registryHost())
+	loginCmd.Stdin, err = authCmd.StdoutPipe()
+	if err != nil {
+		return fmt.Errorf("creating STDOUT pipe: %w", err)
+	}
+	if err := loginCmd.Start(); err != nil {
+		return fmt.Errorf("starting login command: %w", err)
+	}
+	if err := authCmd.Run(); err != nil {
+		return fmt.Errorf("running print-access-token command: %w", err)
+	}
+	if err := loginCmd.Wait(); err != nil {
+		return fmt.Errorf("waiting for login command: %w", err)
+	}
+	return nil
+}
+
+// Setup performs required setup for the helm AR provider.
+// This requires setting up authentication for the helm CLI.
+func (a *ArtifactRegistryHelmProvider) Setup() error {
+	if err := a.registryLogin(); err != nil {
+		return err
+	}
+	return a.createRepository()
+}
+
+// PushURL returns a URL for pushing images to the remote registry.
+// The name parameter is ignored because helm CLI appends the chart name to the image.
+func (a *ArtifactRegistryHelmProvider) PushURL(_ string) (string, error) {
+	return fmt.Sprintf("oci://%s", a.repositoryAddress()), nil
```

Comment:
if this is the same value, you can just call SyncURL here i guess, rather than repeating the format.

Reply:
done